### DISCUSSION TITLE
Using vue-query 'useQuery' hooks from composable inside nuxt middlewares

### DISCUSSION THREAD

**Discussion by Xezard** [nuxt/nuxt#23765] [Questions]
Recently, I came across a situation where I need to make a request in middleware to check the user's moderation status. TanStack vue-query is actively used on the project, all useQuery hooks are wrapped in separate composables. When i'm trying to use method that calls useQuery from these composables in middleware - everything works, but 2 warnings are showed up in browser and server console:
vue-query composables like "uesQuery()" should only be used inside a "setup()" function or a running effect scope. They might otherwise lead to memory leaks.
[Vue warn] on Scope Dispose() is called when there is no active effect scope to be associated with.
Simple code example:
```typescript
function findById(
  payload: Ref<ModerationPayload>,
  options?: UseQueryOptions<Moderation, unknown, Moderation, (string | Ref<ModerationGetPayload>)[]>
): UseQueryReturnType<Moderation, unknown> {
  const fetch = () => {
    const url = MODERATION_ROUTE.replace('{id}', String(payload.value.id))
    return requestToBody<Moderation>(url)
  }

  return useQuery([KEYS.MODERATIONS, payload], fetch, options)
}
```
```typescript
export default defineNuxtRouteMiddleware(async (to, from) => {
  const { findById } = useModerations()
  const payload = computed(() => {
    return {
      id: 1
    } as ModerationPayload
  })

  const { data } = findById(payload)
  console.log(data.value)
}
```
How can this error be avoided? Is there any other way to solve my problem more effectively with avoiding the occurrence of an error?

**Comment by Hebilicious** [comment]
How are you installing vue-query in your app ? Are you using https://nuxt.com/modules/vue-query ?

**Reply by Xezard** [reply]
No, I did not install a separate module for this - there is no need for this. My whole vue-query setup boils down to:
```ts
import { defineNuxtPlugin, useState } from '#app'
import { VueQueryPlugin, QueryClient, hydrate, dehydrate } from '@tanstack/vue-query'

import type { DehydratedState } from "@tanstack/vue-query"

export default defineNuxtPlugin((nuxt) => {
  const vueQueryState = useState<DehydratedState | null>('vue-query')

  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
        staleTime: 5000,
        retry: false,
        suspense: true
      }
    }
  })

  const options = { queryClient }

  nuxt.vueApp.use(VueQueryPlugin, options)

  if (process.server) {
    nuxt.hooks.hook('app:rendered', () => {
      vueQueryState.value = dehydrate(queryClient)
    })
  }

  if (process.client) {
    nuxt.hooks.hook('app:created', () => {
      hydrate(queryClient, vueQueryState.value)
    })
  }
})
```

**Reply by Hebilicious** [reply]
That's what the [module does for you](https://github.com/Hebilicious/vue-query-nuxt/blob/main/packages/vue-query-nuxt/src/runtime/plugin.ts), as well as giving you optional auto-imports for the vue-query composables.

Regarding that warning, I'm not sure why that's happening. 
Would you mind sharing a reproduction? You can also open an issue at https://github.com/Hebilicious/vue-query-nuxt if that's convenient for you

**Reply by pohnean** [reply]
Hi, I did some digging and found that the @tanstack/vue-query uses the getCurrentScope() from 'vue-demi', which is incorrectly returning `undefined` in a nuxt SSR environment. If used we use getCurrentScope from 'vue', it shows that the scope is actually there.

So this warning can be safely ignored. However, it will be nice to get rid of this warning.
